### DISCUSSION TITLE
[Feature request] Allow `enforceExistence` in `folder-structure` to use a `ruleId`

### DISCUSSION THREAD

**Discussion by rishitank** [Igorkowalski94/eslint-plugin-project-structure#19] [Ideas] [answered]
**Is your feature request related to a problem? Please describe.**
If a specific file is present and I need to enforce the existence of another file, I often have to repeat the pattern like `{kebab-case}.styled.jsx` or something like that.

**Describe the solution you'd like**
It would be great if we could provide a `ruleId` to the `enforceExistence` array, e.g.:
```
"enforceExistence": [
    { "ruleId": "styled-file" }
]
...
"styled-file": {
    "name": "{kebab-case}.styled.jsx"
}
```

**Describe alternatives you've considered**
For now I am repeating patterns that already exist as rules, which makes it less maintainable.

**Additional context**

**Comment by Igorkowalski94** [comment]
The `enforceExistence` requires a specific name, not a pattern. It must be the exact name of the file/folder that should exist, and it cannot contain any regex. Therefore, `ruleId` and case references cannot be used within `enforceExistence`.

**Reply by Igorkowalski94** [reply, marked as answer]
In version 2.2.3, I'll also add the ability to change the case for `enforceExistence`, such as `nodeName`, `NodeName`, `node-name`, `node_name`, and `NODE_NAME`. Additionally, I'll include validation to ensure that the correct reference is being used.


### `enforceExistence?: string[]` <a id="enforce-existence"></a>

Enforce the existence of other folders/files when a given folder/file exists.

In `enforceExistence`, the following references are available:

- `{nodeName}` - Take the name of the current file or folder and change it to `camelCase`.
- `{NodeName}` - Take the name of the current file or folder and change it to `PascalCase`.
- `{node-name}` - Take the name of the current file or folder and change it to `kebab-case`.
- `{node_name}` - Take the name of the current file or folder and change it to `snake_case`.
- `{NODE_NAME}` - Take the name of the current file or folder and change it to `SNAKE_CASE`.

> [!WARNING]
> Folder needs to contain at least one file/subfolder with file to be validated. ESLint and Git ignore empty folders, so they won’t be pushed to the repository and will only remain visible locally.

```jsonc
{
  "structure": {
    // If root directory exists.
    "enforceExistence": [
      "src", // ./src must exist.
    ],
    "children": [
      { "name": "*" },
      {
        "name": "src",
        "children": [
          { "name": "stories", "children": [{ "name": "{camelCase}.stories.tsx" }] },
          { "name": "{PascalCase}.test.tsx" },
          {
            // If ./src/ComponentName.tsx exist:
            "name": "{PascalCase}.tsx",
            "enforceExistence": [
              "{NodeName}.test.tsx", // ./src/ComponentName.test.tsx must exist.
              "stories/{nodeName}.stories.tsx", // ./src/stories/componentName.stories.tsx must exist.
              "../cats.ts", // ./cats.ts must exist.
            ],
          },
        ],
      },
    ],
  },
}
```
